vcGL/src/vcTextureHelper.cpp

```diff
@@ -414,3 +415,62 @@ udResult vcTexture_SaveImage(vcTexture *pTexture, vcFramebuffer *pFramebuffer, c
 
   return result;
 }
+
+
+uint8_t *vcTexture_DecodePixels_stbi(void *pFileContents, size_t fileLength, vcImageData *pImageData = nullptr)
+{
+  if (pFileContents == nullptr)
+    return nullptr;
+
+  vcImageData tempData = {};
+  uint8_t *pData_stbi = nullptr;
+  uint8_t *pPixelData = nullptr;
+
+  pData_stbi = stbi_load_from_memory((stbi_uc *)pFileContents, (int)fileLength, (int *)&tempData.width, (int *)&tempData.height, (int *)&tempData.channelCount, 4);
+  if (pData_stbi != nullptr)
+  {
+    pPixelData = (uint8_t *)udMemDup(pData_stbi, (size_t)tempData.width * tempData.height * 4, 0, udAF_None);
+    stbi_image_free(pData_stbi);
+    if (pImageData != nullptr)
+      *pImageData = tempData;
+  }
+
+  return pPixelData;
+}
+
+uint8_t *vcTexture_DecodePixels_tiff(void *pFileContents, size_t fileLength, vcImageData *pImageData = nullptr)
+{
+  if (pFileContents == nullptr)
+    return nullptr;
+
+  vcImageData tempData = {};
+  uint8_t *pPixelData = nullptr;
+
+  if (vcTiff_LoadFromMemory((uint8_t *)pFileContents, fileLength, &tempData.width, &tempData.height, &pPixelData) == udR_Success)
+  {
+    tempData.channelCount = 4;
+    if (pImageData != nullptr)
+      *pImageData = tempData;
+  }
+
+  return pPixelData;
+}
+
+uint8_t *vcTexture_DecodePixels(void *pFileContents, size_t fileLength, vcImageData *pImageData)
+{
+  uint8_t *pPixelData = nullptr;
+
+  do
+  {
+    pPixelData = vcTexture_DecodePixels_stbi(pFileContents, fileLength, pImageData);
+    if (pPixelData != nullptr)
+      break;
+
+    pPixelData = vcTexture_DecodePixels_tiff(pFileContents, fileLength, pImageData);
+    if (pPixelData != nullptr)
+      break;
+
+  } while (false);
+
+  return pPixelData;
```

Comment:
```suggestion
  uint8_t *pPixelData = nullptr;

  // try stbi
  if (pPixelData == nullptr)
    pPixelData = vcTexture_DecodePixels_stbi(pFileContents, fileLength, pImageData);

  // if nothing loaded, try libtiff
  if (pPixelData == nullptr)
    pPixelData = vcTexture_DecodePixels_tiff(pFileContents, fileLength, pImageData);
    
  return pPixelData;
```

Is a bit cleaner and